### PR TITLE
SG-23338-v1

### PR DESCRIPTION
**Pull request by shaynacohen**
added content top to bottom for troubleshooting

### CODE REVIEW

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/e46ca668bb5b09a123f95eb2514ce2dcd7ade5c9/index.html)

**Comment by shaynacohen** [comment]
@drewavisautodesk these are all the files and images associated with linking from the troubleshooting guide. The Troubleshooting guide link updates (to these files) are coming in another PR.

File: docs/en/quick-answers/troubleshooting/ascii-error-message.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+layout: default
+title: ascii codec can’t decode byte 0x97 in position 10
+pagename: ascii-error-message
+lang: en
+---
+
+# ascii codec can’t decode byte 0x97 in position 10: ordinal not in range
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/ascii-error-message.md:8]: Should "ascii" be "ASCII"?

File: docs/en/quick-answers/troubleshooting/ascii-error-message.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Related error messages:
+While cloning a configuration
+- TankError: Could not create file system structure: ‘ascii’ codec can’t decode byte 0x97 in position 10: ordinal not in range(128)
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/ascii-error-message.md:12]: I would replace curly brackets with straight ones, they will probably render better.

File: docs/en/quick-answers/troubleshooting/certificate-fail-local-error-message.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Getting CERTIFICATE_VERIFY_FAILED when using {% include product %} Desktop on a local {% include product %} site
+
+## Use cases:
+When using a local install of {% include product %}, this error can arise in two scenarios:
+
+- when logging in {% include product %} Desktop
+- when downloading media from the Toolkit AppStore
+
+## How to fix:
+To solve this issue, need to provide a file to the {% include product %} API that contains the list of all valid CAs, including your own. We usually recommend that people download a fresh copy of [this file](https://github.com/certifi/python-certifi/blob/master/certifi/cacert.pem) from Python's `certifi` package as a starting point and then add their own CA at the end of the file. Then, save that file in a location all your users can access. Finally, on each computer, set the `SHOTGUN_API_CACERTS` environment variable to the full path to that file, for example `/path/to/my/ca/file.pem`.
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/certificate-fail-local-error-message.md:17]: To solve this issue, *you* need to provide... (missing "you")

File: docs/en/quick-answers/troubleshooting/encryption-handshake-error-message.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+```
+[    INFO] Connection accepted.
+[   ERROR] Attempted to communicate without completing encryption handshake.
+[   ERROR] Attempted to communicate without completing encryption handshake.
+[    INFO] Connection closed.
+[   DEBUG] Reason received for connection loss: [Failure instance: Traceback (failure with no frames): : Connection to the other side was lost in a non-clean fashion: Connection lost.
+```
+
+Opening a project in shotgun produces the following log output:
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/encryption-handshake-error-message.md:39]: Need include product here

File: docs/en/quick-answers/troubleshooting/hiero-task-type-error-message.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+## Use case:
+After updating to `config_default2`, nuke_studio does not initialize. Under Nuke 12.0 studio I get no errors in the script editor, but in Nuke 11.1v3 I get:
+
+```
+ERROR 18:13:28.365:Hiero(34236): Error! Task type tk_hiero_export.sg_shot_processor.ShotgunShotProcessor Not recognised
+```
+
+Without fail after a rollback, it still does not initialize the tk-nuke engine, and {% include product %} fails to load anything…
+
+I have attached a full log, this is in a testing environment, so no worries with NDAs…
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/hiero-task-type-error-message.md:19]: Maybe remove this sentence?

File: docs/en/quick-answers/troubleshooting/invalid-lut-error-message.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+## Use case:
+While working in an ACES color management project, when you use the default toolkit publishing, it fails with an error  `Invalid LUT selected : Gamma2.2`.
+
+## What’s causing the error?
+The issue you are seeing is because there is an app that creates the quicktime that is a part of the toolkit publishing from Nuke called  `tk-multi-reviewsubmission`,  and by default it will create a QT that works with Nukes standard color model.
+
+## How to fix
+Since you are using ACES (I'm assuming the ICIO model), we just need to change the colorspace setting in the  `tk-multi-reviewsubmission`  app by taking over and adding it into the  `codec_settings.py`  hook.
+
+I'm not sure what codec would work best for your usage, but I'm using the  `Output - sRGB Codec`  as an example, you may need to change based on your color workflow. So in the  `codec_settings.py` hook add the setting  `settings["colorspace"] = "Output - sRGB"`  to where it makes sense for your setup. (I've just added it everywhere)
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/invalid-lut-error-message.md:19]: Could be re-worded?

File: docs/en/quick-answers/troubleshooting/invalid-lut-error-message.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+lang: en
+---
+
+# error Invalid LUT selected : Gamma2.2
+
+## Use case:
+While working in an ACES color management project, when you use the default toolkit publishing, it fails with an error  `Invalid LUT selected : Gamma2.2`.
+
+## What’s causing the error?
+The issue you are seeing is because there is an app that creates the quicktime that is a part of the toolkit publishing from Nuke called  `tk-multi-reviewsubmission`,  and by default it will create a QT that works with Nukes standard color model.
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/invalid-lut-error-message.md:14]: reword?

File: docs/en/quick-answers/troubleshooting/path-associated-error-message.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+## Related error messages:
+
+- Database concurrency problems: The path `<PATH>` is already associated with {% include product %} entity `<ENTITY>`.
+- Could not resolve row id for path!
+
+## Example:
+This error occurs when a Toolkit user tries to create folders. Here’s the full error:
+
+The full error looks something like:
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/path-associated-error-message.md:18]: Remove this line?

File: docs/en/quick-answers/troubleshooting/paths-long-error-message.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+ERROR sgtk.core.bootstrap.cached_configuration Failed to install configuration sgtk:descriptor:shotgun?entity_type=PipelineConfiguration&field=sg_uploaded_config&id=1&version=1182. Error: Failed to download into path C:\Users\xxxxx\AppData\Roaming\Shotgun\bundle_cache\tmp\dd2cc0804122403a87ac71efccd383ea: Failed to download sgtk:descriptor:shotgun?entity_type=PipelineConfiguration&field=sg_uploaded_config&id=1&version=1182 from https://xxxxx.shotgunstudio.com. Error: Failed to download from 'https://xxxxx.shotgunstudio.com' after 5 retries. See error log for details.. Cannot continue.
+```
+
+## Why this happens
+On Windows, SG Desktop stores data in your `%APPDATA%` folder (typically `C:\Users\jane\AppData\Roaming\Shotgun`. When using the standard default2 Toolkit config things should be mostly fine as long as your username isn't super long. However, if you are creating your own apps, engines, or frameworks, you may have more risk of running into this, especially if you bundle dependencies with your code (as we do), and you have deep trees of directories in your bundles.
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/paths-long-error-message.md:33]: Include product name?

File: docs/en/quick-answers/troubleshooting/publish-certificate-fail-error-message.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+ShotgunPublishError: Unable to complete publishing because of the following error: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726)>, although PublishedFile PIGS_libary_tool_hda_asasas_v017.hda (id: 114715
+) was created.
+---------------------------------------------------------------------------
+```
+
+## What’s causing the error?
+Missing `cacert.pem` and the required environment variable `SHOTGUN_API_CACERTS` pointing to that location.
+
+## How to fix
+Add tje `cacert.pem` and environment variable `SHOTGUN_API_CACERTS` pointing to the location.
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/publish-certificate-fail-error-message.md:70]: tje?

File: docs/en/quick-answers/troubleshooting/review-submission-error-message.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    
+    _handle_plugin_error
+        yield
+
+    run_publish
+        self._hook_instance.publish(settings, item)
+
+I can find these in my local Appdata folder but not in my main install so I'm assuming that I need to look in 'submit_for_review.py' to get at this problem.
+
+Looking at this file I can see that its sucking settings and info for the item from higher up the hereditary chain 'HookBaseClass'.
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/review-submission-error-message.md:53]: Maybe reword "sucking"?

File: docs/en/quick-answers/troubleshooting/review-submission-error-message.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+So looking for where the 'sg_publish_data' is set in 'submit_for_review.py' I see line 225:
+`sg_publish_data = item.properties.get("sg_publish_data")`
+
+**question 2**
+So where upstream is 'item.properties' is being set? I suspect in the nuke collector.py but 'sg_publish_data' is not present in there and is also a child of the 'HookBaseClass'
+
+## How to fix
+
+### The Short Answer:
+Assuming you have not added custom code or modified the config too much, you just need to make sure you have `Publish to Shotgun` checked on the item that you are  submitting for review
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/review-submission-error-message.md:64]: include product?

File: docs/en/quick-answers/troubleshooting/review-submission-error-message.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+The idea for the combination of these two plugins is that you are submitting for review something that is published (and will not change). So the workflow is to publish the source files first, then create a SG Version from them for review. The version is linked to the publish as well. the first plugin does the file publish and stores the information in the item for the second publish to use.  
+
+You mentioned inheritance so I wanted to clarify that as well. I think what you're coming across in the documentation is referring to hook inheritance that is driven by the config and not a python inheritance as I suspect you think it means. 
+
+In the python files you'll see that all the hooks inherit from `HookBaseClass`. But the hook mechanism in the config allows you to essentially treat these base hooks as mixins and combine them to get all the functionality you need while reusing code. You can see an example of this in the `tk-multi-publish2` config:
+
+```
+settings.tk-multi-publish2.nuke.shot_step:
+  collector: "{self}/collector.py:{engine}/tk-multi-publish2/basic/collector.py"
+```
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/review-submission-error-message.md:106]: add blank line

File: docs/en/quick-answers/troubleshooting/row-id-error-message.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
+---
+layout: default
+title: Could not resolve row id for path!
+pagename: row-id-error-message
+lang: en
+---
+
+# Could not resolve row id for path!
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/row-id-error-message.md:8]: exclamation mark?

File: docs/en/quick-answers/troubleshooting/tk-maya-exception-error-message.md [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+
+For example, when you run an app, you may want to provide some kind of state flag which causes the app to launch differently depending on the state.
+
+Here are a couple of examples of where this is already used:
+
+- The `tk-shotgun-folders` app, (which creates folders based on the selected entities in the Shotgun web app,) will get passed the Shotgun entity/s and the entity type that the user selected in the Shotgun web app and ran the action on:
+https://github.com/shotgunsoftware/tk-shotgun-folders/blob/v0.1.7/app.py#L86
+- The `tk-multi-launchapp` (which is responsible for launching software with the Shotgun integrations) can be passed a `file_to_open` arg, which it will then use to open the file once the software is launched:
+https://github.com/shotgunsoftware/tk-multi-launchapp/blob/v0.11.2/python/tk_multi_launchapp/base_launcher.py#L157
+Normally when you launch the software through Shotgun Desktop it won't provide a `file_to_open` argument, however, you can call the app via the tank command if you are using a centralized config (`tank maya_2019 /path/to/maya/file.mb`).
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/tk-maya-exception-error-message.md:22]: include product?

File: docs/en/quick-answers/troubleshooting/unreal-proxy-error-message.md [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+File “C:\Users\USER0\AppData\Roaming\Shotgun\bundle_cache\app_store\tk-multi-launchapp\v0.10.2\python\tk_multi_launchapp\base_launcher.py”, line 125, in launch_version
+*args, **kwargs
+File “C:\Users\USER0\AppData\Roaming\Shotgun\bundle_cache\app_store\tk-multi-launchapp\v0.10.2\python\tk_multi_launchapp\base_launcher.py”, line 343, in _launch_callback
+“Could not create folders on disk. Error reported: %s” % err
+TankError: Could not create folders on disk. Error reported: Could not resolve row id for path! Please contact support! trying to resolve path ‘D:\UEProjects\SON\D:\UEProjects\SON’. Source data set: [{‘path_cache_row_id’: 2, ‘path’: ‘D:\UEProjects\SON’, ‘metadata’: {‘root_name’: ‘primary’, ‘type’: ‘project’}, ‘primary’: True, ‘entity’: {‘type’: ‘Project’, ‘id’: 91, ‘name’: ‘SON’}}]
+
+```
+
+## What’s causing the error?
+The path of the UE4’s Project on disk which was `D:\UEProjects\PROJECT_NAME\ incorrect.
```
Comment (drewavisautodesk) [inline docs/en/quick-answers/troubleshooting/unreal-proxy-error-message.md:29]: missing backtick

**Review by drewavisautodesk — APPROVED** [review]
Hi Shayna, leaving some comments, but overall looks good.  One style comment: there should be a blank line in markdown after headings and code blocks.

File: docs/en/quick-answers/troubleshooting/ascii-error-message.md (hunk 1 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/ascii-error-message.md:8]: Fixed!

File: docs/en/quick-answers/troubleshooting/ascii-error-message.md (hunk 2 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/ascii-error-message.md:12]: Fixed

File: docs/en/quick-answers/troubleshooting/certificate-fail-local-error-message.md (hunk 3 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/certificate-fail-local-error-message.md:17]: Thank you! Fixed!

File: docs/en/quick-answers/troubleshooting/encryption-handshake-error-message.md (hunk 4 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/encryption-handshake-error-message.md:39]: Thanks! Nice catch :)

File: docs/en/quick-answers/troubleshooting/hiero-task-type-error-message.md (hunk 5 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/hiero-task-type-error-message.md:19]: Thanks!

File: docs/en/quick-answers/troubleshooting/invalid-lut-error-message.md (hunk 7 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/invalid-lut-error-message.md:14]: Fixed, thanks!

File: docs/en/quick-answers/troubleshooting/invalid-lut-error-message.md (hunk 6 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/invalid-lut-error-message.md:19]: Fixed, thanks!

File: docs/en/quick-answers/troubleshooting/path-associated-error-message.md (hunk 8 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/path-associated-error-message.md:18]: Yes, good catch! Thank you!

File: docs/en/quick-answers/troubleshooting/paths-long-error-message.md (hunk 9 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/paths-long-error-message.md:33]: Fixed, thanks!

File: docs/en/quick-answers/troubleshooting/publish-certificate-fail-error-message.md (hunk 10 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/publish-certificate-fail-error-message.md:70]: Thanks!

File: docs/en/quick-answers/troubleshooting/review-submission-error-message.md (hunk 11 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/review-submission-error-message.md:53]: Facepalm... thanks!

File: docs/en/quick-answers/troubleshooting/review-submission-error-message.md (hunk 12 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/review-submission-error-message.md:64]: Thanks!

File: docs/en/quick-answers/troubleshooting/review-submission-error-message.md (hunk 13 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/review-submission-error-message.md:106]: Thanks!

File: docs/en/quick-answers/troubleshooting/tk-maya-exception-error-message.md (hunk 15 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/tk-maya-exception-error-message.md:22]: Thanks!

File: docs/en/quick-answers/troubleshooting/unreal-proxy-error-message.md (hunk 16 above)
Comment (shaynacohen) [reply docs/en/quick-answers/troubleshooting/unreal-proxy-error-message.md:29]: Thanks!

**Comment by shaynacohen** [comment]
Thanks @drewavisautodesk ! I fixed up all the spacing, too!

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/32819110056902c9414c23c9cc04b4dd17d671fe/index.html)
